presto-main/src/main/java/io/prestosql/metadata/QualifiedTablePrefix.java

```diff
@@ -44,15 +43,15 @@ public QualifiedTablePrefix(String catalogName)
     public QualifiedTablePrefix(String catalogName, String schemaName)
     {
         this.catalogName = checkCatalogName(catalogName);
-        this.schemaName = Optional.of(checkSchemaName(schemaName));
+        this.schemaName = Optional.of(requireNonNull(schemaName, "schemaName is null"));
```

Comment:
`Optional.of` will validate that the argument is not null, so the `requireNonNull` is unnecessary.